src/solvers/bicgstab_solver.hpp

```diff
@@ -262,11 +251,9 @@ class BiCGSTABSolver {
           solver->residual.val = 0.0;
           return TaskStatus::iterate;
         },
-        this, pmesh, i, params_.max_iters, params_.residual_tolerance);
-    region.AddGlobalDependencies(reg_dep_id, i, check);
-    reg_dep_id++;
+        this, pmesh, params_.residual_tolerance);
 
-    return check;
+    return solver_id;
```

Comment:
Why aren't we returning `check`? If this is the new and correct API, I am not sure if I like it, since `check` is actually where we are assigning `TaskStatus::iterate` vs `TaskStatus::complete`.

Reply:
An iteration can now have more than one completion task.  The cleanest way I could come up with to support this kind of thing was to have an ID associated with the whole sublist (in this case solver_id).  So tasks that depend on an iteration being complete now depend on the iterative list being complete instead of a particular task within it.  I know what you mean, but I'd actually argue this is cleaner and clearer.  Dependent tasks don't depend on the completion task, they depend on the iteration being complete, whatever that means for the iteration you're trying to write (which may have multiple places where you might exit).